buildSrc/build.gradle

```diff
@@ -96,9 +90,25 @@ dependencies {
   compile 'com.perforce:p4java:2012.3.551082' // THIS IS SUPPOSED TO BE OPTIONAL IN THE FUTURE....
   compile 'de.thetaphi:forbiddenapis:2.2'
   compile 'org.apache.rat:apache-rat:0.11'
-  compile 'ru.vyarus:gradle-animalsniffer-plugin:1.0.1'
 }
 
+// Gradle version-specific options (allows build to run with Gradle 2.13 as well as 2.14+/3.+)
+if (GradleVersion.current() == GradleVersion.version("2.13")) {
+  // ProgressLogger(-Factory) classes are part of the public Gradle API
+  // Add default imports for (org.gradle.logging)
+  compileGroovy.groovyOptions.configurationScript = file('src/compile/groovy/gradle-2.13-loggers.groovy')
+  dependencies {
+    compile 'ru.vyarus:gradle-animalsniffer-plugin:1.0.1' // last version compatible with Gradle 2.13
+  }
+} else {
+  // Gradle 2.14+ removed ProgressLogger(-Factory) classes from the public APIs
+  // Use logging dependency and add default imports for "org.gradle.internal.logging.progress"
+  compileGroovy.groovyOptions.configurationScript = file('src/compile/groovy/gradle-2.14-loggers.groovy')
+  dependencies {
+    compile "org.gradle:gradle-logging:3.3" // any version >= 2.14 is ok, take a recent one
```

Comment:
The reason I fixed the version is because we export gradle-logging as dependency when publishing `build-tools`. This would mean that the dependencies of `build-tools` would change depending on whether someone uses Gradle 3.2 or 3.3 to build the artifact (the issue is still there between 2.13 and 2.14, but if we would move to 2.14+ in the near future, we could require 2.14 as minimal version and at least avoid this discrepancy from that point onward). Even though we use a fixed Vagrant image for building (with specific Gradle version installed), I felt that this would be too brittle. Another possibility to reuse the ProgressLogger code might be to shade the dependency, which could make it uniformly available in Gradle 2.13 and 2.14+. We would then have to publish that shaded artifact as well though.